### ISSUE TITLE
Increase text-type Custom Field length to accommodate Azure Resource IDs

### ISSUE THREAD

**Issue by kevinsumner** [nautobot/nautobot#5228]
### As ...

P.D. - Plugin Developer

### I want ...

to be able to store an Azure Resource IDs whose lengths are over 255 characters in a custom field, in both 1.6 and 2.x.

For example, this is the generic form of a [subnet's ID in Azure's REST API](https://learn.microsoft.com/en-us/rest/api/virtualnetwork/subnets/get):

``` /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/subnets/{subnetName}
```

In the above, `resourceGroupName`, `virtualNetworkName`, and `subnetName` are arbitrary strings created by an Azure admin, thus can have arbitrary length. 

There isn't a shorter, globally- or subscription-unique identifier I can find in Azure's API, such as an object UUID or something like AWS' IDs (e.g. `subnet-xxxxxxxxxxxxxxxx` for subnets in AWS).

Further reading on Azure resource naming: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules

### So that ...

my SSOT plugin may use the resource ID stored in the custom field to reference back to the Azure API object to allow DiffSync to locate the associated records.

### I know this is done when...

my plugin can use our Azure resources whose IDs are larger than 255 characters to lookup objects in Nautobot that originated in Azure, and to use the Azure resource ID on a given object in Nautobot to reference back to the canonical object in Azure's API. Currently, that's at least some Azure Virtual Networks' subnets (mapped to IP Prefixes for us), although I'm still investigating what other Azure resource types have similar concerns.

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [X] Misc (including Data Sources)
- [ ] Organization
- [X] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

Based on the [1.6 documentation of Custom Fields](https://docs.nautobot.com/projects/core/en/v1.6.10/user-guides/custom-fields/) and a cursory look at `nautobot/extras/models/customfields.py` on main branch, I believe the 255 character limit is imposed by the existing schema.

**Comment by glennmatthews** [comment]
Related to but distinct from #2906

**Comment by kevinsumner** [comment]
Are those fields searchable? In other words, can I use them to pull a specific object by the custom field's value?

**Comment by glennmatthews** [comment]
Yes, they're usable as filters in the UI and REST API, as well as in the ORM  - here it is as a Markdown custom field:

![image](https://github.com/nautobot/nautobot/assets/5603551/666a7602-83e9-40c8-aafc-5fcdc0d5e0e0)

**Comment by lampwins** [comment]
Probably makes sense to make this an attribute of a custom field definition. Make it optional, if not set, no limit, otherwise enforced.

**Comment by lampwins** [comment]
Let's reuse the min and max value fields that already exist on the custom field model to support numeric and text type fields. Leave the default null values for the fields, and only enforce them when a value is specified.

<img width="1120" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMnZOQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--86ba9d505d57a293eed8376582aa43078b5f5a5b/Screenshot%202024-03-11%20at%201.18.01%20PM.png" alt="Screenshot 2024-03-11 at 1.18.01 PM.png" />

**Comment by nrnvgh** [comment]
Any chance of getting this backported to `ltm-1.6`?
